nemo_rl/models/generation/vllm/routing_policy.py

```
@@ -0,0 +1,246 @@
+        *,
+        data: BatchedDataDict,
+        request_id: str,
+    ) -> RoutingDecision:
+        decision = self._external_router.route(data)
+
+        dp_size = len(self._dp_leader_worker_indices)
+        if not (0 <= decision.dp_shard_idx < dp_size):
+            raise ValueError(
+                f"External router returned dp_shard_idx={decision.dp_shard_idx}, "
+                f"expected range [0, {dp_size})"
+            )
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)